docs/platforms/android/Kotlin-formatting.md

```diff
@@ -0,0 +1,27 @@
+# Setting up Kotlin formatting in Android Studio
+
+Kotlin code in this repository is formatted and linted with [`ktlint`](https://github.com/pinterest/ktlint).
+If you
+1. have submitted Kotlin code only to learn this when the analyzer check fails
+2. use Android Studio
+
+
+then there is good news! Android Studio can be configured to use `ktlint` to automatically apply formatting and highlight issues. To do this:
+
+
+1. Install the `ktlint` extension for Android Studio
+
+   a. On Mac, this is `Android Studio > Settings > Plugins > ` Search for `ktlint`.
+
+2. Set the ruleset to be the same as the version used in `ci.yaml` (as of writing this is 1.1.1), and the baseline to be `dev/bots/test/analyze-test-input/ktlint-baseline.xml`.
+
+   a. Both of these options should be available under `Android Studio > Settings > Tools > ktlint`.
+
+3. Additionally, Kotlin code in the Flutter repository currently uses some additional rules for compatibility with older versions of Kotlin.
+These rules can only be configured by an `.editorconfig` file in the directory from which Android Studio was opened. To configure these rules, create a new `.editorconfig` file in the root of your flutter repository with the following content
+```
```

Comment:
Not required to merge this pr but I would encourage you to make a .editorconfig file in the repo then have the instructions here be to set that path as the file that should be used. That way if we need to update it everyone who has linked the path gets the new rules.

Reply:
I think this would make sense. Right now I've been preserving a gitignored .editorconfig file which isn't the best solution. I can do this in a follow up